lib/iris/fileformats/netcdf.py

```diff
@@ -393,6 +393,10 @@ def __getitem__(self, keys):
             variable = dataset.variables[self.variable_name]
             # Get the NetCDF variable data and slice.
             var = variable[keys]
+            # Make into a 'plain' array if no masked points,
+            # for compatibility with netcdf <= 1.3.
+            if isinstance(var, np.ma.MaskedArray) and not np.ma.is_masked(var):
```

Comment:
@pp-mo I assume that you know what you're doing, but just for due diligence can you please confirm that `np.ma.is_masked` will check whether there are any masked points, rather than whether the array has a mask?

Reply:
It does, I checked !  
For some absolute pedantic certainty ...
```
>>> test_arrays = (('normal', np.arange(3)),
...                ('mask_false', np.ma.masked_array([1, 2])),
...                ('mask_empty', np.ma.masked_array([1, 2], mask=[0, 0])),
...                ('masked_points', np.ma.masked_array([1, 2], mask=[0, 1])))
>>> for name, array in test_arrays:
...     mask = array.mask if isinstance(array, np.ma.MaskedArray) else None
...     print('{:15s} :  mask={!s:15s} isMaskedArray={!s:5s} is_masked={!s:5s}'.format(name, mask, np.ma.isMaskedArray(array), np.ma.is_masked(array)))
... 
normal          :  mask=None            isMaskedArray=False is_masked=False
mask_false      :  mask=False           isMaskedArray=True  is_masked=False
mask_empty      :  mask=[False False]   isMaskedArray=True  is_masked=False
masked_points   :  mask=[False  True]   isMaskedArray=True  is_masked=True 
>>> 
```

Reply:
Awesome.  I'm convinced.